server/aesdsocket.c

```diff
@@ -107,18 +174,289 @@ static int create_daemon()
         return 0;
 }
 
+static void initTimer(void)
+{
+    int timer_rc;
+    pthread_t timertid;
+    if((timer_rc = pthread_create(&timertid, NULL, threadtimerfunc, NULL)) != 0)
+    {
+        syslog(LOG_ERR, "Failed to pthread_create() timer_thread, error was %d", timer_rc);
+        perror("Failed to pthread_create() timer_thread");
+    }
+    syslog(LOG_DEBUG, "Timer thread created successfully");
+    return;
+}
+
 
-void closeAll()
+
+void closeAll(int exit_flag)
 {
+    //int rc;
     syslog(LOG_DEBUG, "PERFORMING CLEANUP.");
     if(sockfd != -1) close(sockfd);
     if(new_fd != -1) close(new_fd);
+    if(recvfile_fd != -1) close(recvfile_fd);
     if(remove(recvfile) != 0)
     {
-        syslog(LOG_ERR, "File removal not successful");
+       syslog(LOG_ERR, "File removal not successful");
     }
-    closelog();
+    syslog(LOG_DEBUG, "PERFORMING CLEANUP1");
+    //timer_delete(timerid);
+    pthread_mutex_destroy(&writeSocket);
```

Comment:
@sota6640 you need to think about whether it's safe to do this cleanup action in each place where the cleanup function is called.  `writeSocket` is one thing which probably isn't safe to destroy in every case you call this function, since your timerThread may still be running.
This is another reason why it's generally better/easier to do cleanup in one place and one code path, much easier to review sequencing.